pkg/caveats/customtypes/customtypes.go

```diff
@@ -0,0 +1,11 @@
+package customtypes
+
+import (
+	"github.com/google/cel-go/cel"
+)
+
+var CustomTypes = map[string][]cel.EnvOption{}
+
+func registerCustomType(name string, opts ...cel.EnvOption) {
```

Comment:
exposing this would allow folks to inject their own types, but it also means exposing CEL as part of the API. I think exploring how to abstract this and export it in a follow up would be a nice addition.

Reply:
Agreed.